tests/test_registry.py

```diff
@@ -645,6 +646,66 @@ def raise_exception():
         assert "error" in response
         assert random_val == response["error"]["data"]["random"]
 
+    def test_encoder_exception_batch(self):
+        registry = Registry()
+        original_encode = registry.json_encoder.encode
```

Comment:
You can implement this with a scoped `pytest.patch`. If you want some advice on actually how to do this, feel free to ask.

Implementing it yourself doesn't handle the case where an exception is thrown in the middle.